python/sidecar/search.py

```diff
@@ -0,0 +1,256 @@
+"""
+Author: Shaurya
+This script performs a search on Semantic Scholar and retrieves information about a set of papers.
+
+Instructions:
+The script takes two command line arguments:
+--query: The term you want to search on Semantic Scholar, written in quotation marks (e.g., "brain computer interface").
+--limit: Optional, the maximum number of papers to retrieve. Default is 10.
+
+Run the script like so from your terminal:
+    python search.py --query "computational linguistics" --limit 10
+"""
+import argparse
+from semanticscholar import SemanticScholar
```

Comment:
We'll need to add this as a dependency via `poetry add semanticscholar` and adding the resulting update to pyproject.toml and poetry.lock.